File: app/src/main/java/nl/acidcats/tumblrlikes/ui/fragments/BaseFragment.java
package nl.acidcats.tumblrlikes.ui.fragments;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.CallSuper;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.content.LocalBroadcastManager;
import android.view.View;

import butterknife.ButterKnife;
import butterknife.Unbinder;
import nl.acidcats.tumblrlikes.LikesApplication;
import nl.acidcats.tumblrlikes.di.AppComponent;

/**
 * Created on 30/07/2018.
 */
public abstract class BaseFragment extends Fragment {

    private Unbinder _unbinder;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        if (getActivity() != null && getActivity().getApplication() != null) {
            injectFrom(((LikesApplication) getActivity().getApplication()).getAppComponent());
        }
    }

    abstract protected void injectFrom(AppComponent appComponent);

    @CallSuper
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        _unbinder = ButterKnife.bind(this, view);
    }

    protected void sendBroadcast(String action) {
        if (getContext() != null) {
            LocalBroadcastManager.getInstance(getContext()).sendBroadcast(new Intent(action));
        }
    }

    @CallSuper
    @Override
    public void onDestroyView() {
        _unbinder.unbind();

        super.onDestroyView();
    }
}

File: app/src/main/java/nl/acidcats/tumblrlikes/ui/screens/base/BaseFragment.kt
package nl.acidcats.tumblrlikes.ui.screens.base

import android.content.Context
import android.content.Intent
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import nl.acidcats.tumblrlikes.di.Injectable

/**
 * Created on 29/10/2018.
 */
abstract class BaseFragment : Fragment(), BaseView, Injectable {

    override fun sendBroadcast(action: String) {
        context?.let {
            LocalBroadcastManager.getInstance(it).sendBroadcast(Intent(action))
        }
    }

    override fun clearArgument(key: String) {
        arguments?.remove(key)
    }

    override fun showToast(message: String?) {
        if (context != null && message != null) {
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
        }
    }

    override fun getContext(): Context? {
        return super.getContext()
    }
}